gateway-ha/src/main/java/io/trino/gateway/ha/router/HaGatewayManager.java

```diff
@@ -110,7 +131,10 @@ public ProxyBackendConfiguration updateBackend(ProxyBackendConfiguration backend
             dao.create(backend.getName(), backend.getRoutingGroup(), backendProxyTo, backendExternalUrl, backend.isActive());
         }
         else {
+            Boolean prevStatus = model.active();
             dao.update(backend.getName(), backend.getRoutingGroup(), backendProxyTo, backendExternalUrl, backend.isActive());
+            Boolean currStatus = backend.isActive();
+            logActivationStatusChange(backend.getName(), currStatus, prevStatus);
```

Comment:
Is it possible that `model.active()` or `backend.isActive()` could return null? If so you will get NPE on L137 since `logActivationStatusChange` accepts `boolean` (primitive type) so unboxing will occur here. So if null is possible, then `logActivationStatusChange` has to accept `Boolean` (boxed type). If it can't be null, all good.

Btw the local variables don't seem necessary:
```suggestion
            dao.update(backend.getName(), backend.getRoutingGroup(), backendProxyTo, backendExternalUrl, backend.isActive());
            logActivationStatusChange(backend.getName(), backend.isActive(), model.active());
```

Reply:
Looks like `.active` and `.isActive` return primitive type `boolean` so I don't think they can be null . We should be good! Thanks for pointing that out